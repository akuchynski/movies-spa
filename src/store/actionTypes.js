export default {

    GET_MOVIES_REQUEST: 'GET_MOVIES_REQUEST',
    GET_MOVIES_SUCCESS: 'GET_MOVIES_SUCCESS',
    GET_MOVIES_ERROR: 'GET_MOVIES_ERROR',

    GET_MOVIE_REQUEST: 'GET_MOVIE_REQUEST',
    GET_MOVIE_SUCCESS: 'GET_MOVIE_SUCCESS',
    GET_MOVIE_ERROR: 'GET_MOVIE_ERROR',

    CLOSE_MOVIE_DETAILS: 'CLOSE_MOVIE_DETAILS',
    SET_ACTIVE_FILTER: 'SET_ACTIVE_FILTER',

    ADD_MOVIE_SUCCESS: 'ADD_MOVIE_SUCCESS',
    ADD_MOVIE_ERROR: 'ADD_MOVIE_ERROR',

    UPDATE_MOVIE_SUCCESS: 'UPDATE_MOVIE_SUCCESS',
    UPDATE_MOVIE_ERROR: 'UPDATE_MOVIE_ERROR',

    DELETE_MOVIE_SUCCESS: 'DELETE_MOVIE_SUCCESS',
    DELETE_MOVIE_ERROR: 'DELETE_MOVIE_ERROR',

};