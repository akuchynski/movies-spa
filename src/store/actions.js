import actionTypes from "./actionTypes";

const getMoviesRequest = () => ({
    type: actionTypes.GET_MOVIES_REQUEST,
});

const getMoviesSuccess = (movies) => ({
    type: actionTypes.GET_MOVIES_SUCCESS,
    payload: movies
});

const getMoviesError = (errorMessage) => ({
    type: actionTypes.GET_MOVIES_ERROR,
    payload: errorMessage
});

const getMovieRequest = () => ({
    type: actionTypes.GET_MOVIE_REQUEST,
});

const getMovieSuccess = (movie) => ({
    type: actionTypes.GET_MOVIE_SUCCESS,
    payload: movie
});

const getMovieError = (errorMessage) => ({
    type: actionTypes.GET_MOVIE_ERROR,
    payload: errorMessage
});

const closeMovieDetails = () => ({
    type: actionTypes.CLOSE_MOVIE_DETAILS,
});

const setActiveFilter = (value) => ({
    type: actionTypes.SET_ACTIVE_FILTER,
    payload: value
});

export default {
    getMoviesRequest,
    getMoviesSuccess,
    getMoviesError,
    getMovieRequest,
    getMovieSuccess,
    getMovieError,
    closeMovieDetails,
    setActiveFilter,
};