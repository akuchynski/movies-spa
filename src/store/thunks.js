import axios from "axios";
import actions from "./actions";

export const getMovies = () => (dispatch) => {
    dispatch(actions.getMoviesRequest());
    axios.get('http://localhost:4000/movies')
        .then((response) => dispatch(actions.getMoviesSuccess(response.data)))
        .catch((error) => dispatch(actions.getMoviesError(error.message)));
};

export const getMoviesByGenre = (genre) => (dispatch) => {
    dispatch(actions.getMoviesRequest());
    axios.get('http://localhost:4000/movies', genre !== 'ALL' && { params: { searchBy: 'genres', filter: genre } })
        .then((response) => dispatch(actions.getMoviesSuccess(response.data)))
        .catch((error) => dispatch(actions.getMoviesError(error.message)));
};

export const getMoviesByDate = () => (dispatch) => {
    dispatch(actions.getMoviesRequest());
    axios.get('http://localhost:4000/movies?sortBy=release_date&sortOrder=desc')
        .then((response) => dispatch(actions.getMoviesSuccess(response.data)))
        .catch((error) => dispatch(actions.getMoviesError(error.message)));
};

export const getMoviesByRating = () => (dispatch) => {
    dispatch(actions.getMoviesRequest());
    axios.get('http://localhost:4000/movies?sortBy=vote_average&sortOrder=desc')
        .then((response) => dispatch(actions.getMoviesSuccess(response.data)))
        .catch((error) => dispatch(actions.getMoviesError(error.message)));
};

export const getMovieById = (id) => (dispatch) => {
    dispatch(actions.getMovieRequest());
    axios.get('http://localhost:4000/movies/' + id)
        .then((response) => dispatch(actions.getMovieSuccess(response.data)))
        .catch((error) => dispatch(actions.getMovieError(error.message)));
};

export const addMovie = (movie) => (dispatch) => {
    movie.genres = movie.genres.map(item => item.value);
    axios.post('http://localhost:4000/movies', movie)
        .then((response) => dispatch(actions.addMovieSuccess(response.data)))
        .catch((error) => dispatch(actions.addMovieError(error.message)));
};

export const updateMovie = (movieId, movie) => (dispatch) => {
    movie.id = movieId;
    movie.genres = movie.genres.map(item => item.value);
    axios.put('http://localhost:4000/movies', movie)
        .then((response) => dispatch(actions.updateMovieSuccess(response.data)))
        .catch((error) => dispatch(actions.updateMovieError(error.message)));
};

export const deleteMovie = (id) => (dispatch) => {
    axios.delete('http://localhost:4000/movies/' + id)
        .then(() => dispatch(actions.deleteMovieSuccess()))
        .catch((error) => dispatch(actions.deleteMovieError(error.message)));
};