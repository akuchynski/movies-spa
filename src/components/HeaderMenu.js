import React, { useEffect } from "react";
import { useSelector, useDispatch } from "react-redux";
import { MovieDetails } from "./Movie/MovieDetails";
import { MovieSearch } from "./Movie/MovieSearch";
import { getMovieById } from "../store/thunks";
import { useSearchParams } from "react-router-dom";

export const HeaderMenu = () => {

    const dispatch = useDispatch();
    const [searchParams] = useSearchParams();
    const movieParam = searchParams.get("movie");
    const { movieDetails } = useSelector(state => state.movies);

    useEffect(() => {
        if (movieParam)
            dispatch(getMovieById(movieParam));
    }, []);

    return (
        movieDetails ? <MovieDetails /> : <MovieSearch />
    );
};