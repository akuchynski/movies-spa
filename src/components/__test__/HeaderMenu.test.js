import React from "react";
import { render } from "@testing-library/react";
import { Provider } from 'react-redux';
import { store } from "../../store/configureStore";
import { HeaderMenu } from "../HeaderMenu";

describe('header menu components tests', () => {
    it('should render header menu component', () => {
        const { asFragment } = render(
            <Provider store={store}>
                <HeaderMenu />
            </Provider>);
        expect(asFragment()).toMatchSnapshot();
    });
});