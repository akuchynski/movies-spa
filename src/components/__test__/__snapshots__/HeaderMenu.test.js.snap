// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`content components tests should render app component 1`] = `
<DocumentFragment>
  <div
    class="header"
  >
    <div
      class="logo"
    >
      movie
      <span>
        roulette
      </span>
    </div>
    <button
      class="add-button"
    >
      + ADD MOVIE
    </button>
    <div
      class="search"
    >
      <h1>
        FIND YOUR MOVIE
      </h1>
      <form>
        <input
          autocomplete="off"
          id="searchQuery"
          name="searchQuery"
          placeholder="What do you want to watch?"
          type="text"
          value=""
        />
        <button
          class="search-button"
          type="submit"
        >
          SEARCH
        </button>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
